examples/cpp/hello_world/README.md

```diff
@@ -2,66 +2,76 @@
+* The wait-set is a mechanism where a dedicated thread waits until a status condition occurs.
+In that moment, that status condition triggering event would be evaluated to determine witch actions should be taken against it.
+
+For this example, both listening callback and wait-set implementation would run similar code and generate equivalent output for both triggering events: subscription matching and new data available.
 
 ## Run the example
 
-To launch this test open two different consoles. One of them will run the publisher side of the example, and the other would run the subscriber side.
+To launch this example, two different terminals are required.
+One of them will run the publisher example application, and the other would run the subscriber application.
 
 ### Hello world publisher
 
-#### Ubuntu ( / MacOS )
+* Ubuntu ( / MacOS )
 
-```shell
-user@machine:example_path$ ./hello_world publisher
-Publisher running. Please press Ctrl+C to stop the Publisher at any time.
-```
+    ```shell
+    user@machine:example_path$ ./hello_world publisher
+    Publisher running. Please press Ctrl+C to stop the Publisher at any time.
+    ```
 
-#### Windows
+* Windows
 
-```powershell
-example_path> hello_world.exe publisher
-Publisher running. Please press Ctrl+C to stop the Publisher at any time.
-```
+    ```powershell
+    example_path> hello_world.exe publisher
+    Publisher running. Please press Ctrl+C to stop the Publisher at any time.
+    ```
 
 ### Hello world subscriber
 
-#### Ubuntu ( / MacOS )
+* Ubuntu ( / MacOS )
 
-```shell
-user@machine:example_path$ ./hello_world subscriber
-Subscriber running. Please press Ctrl+C to stop the Subscriber at any time.
-```
+    ```shell
+    user@machine:example_path$ ./hello_world subscriber
+    Subscriber running. Please press Ctrl+C to stop the Subscriber at any time.
+    ```
 
-#### Windows
+* Windows
 
-```powershell
-example_path> hello_world.exe subscriber
-Subscriber running. Please press Ctrl+C to stop the Subscriber at any time.
-```
+    ```powershell
+    example_path> hello_world.exe subscriber
+    Subscriber running. Please press Ctrl+C to stop the Subscriber at any time.
+    ```
+
+All the example available flags can be queried running the executable with the ``-h`` or ``--help`` flag.
 
 ## Expected output
 
-It does not matter which entity is launched first, because the publisher has a condition to wait until the first match to start sending hello world samples.
-The expected output on both cases is a first displayed message acknowledging they have matched, followed by the amount of samples sent or received until user press Ctrl+C.
+Regardless of which application is run first, since the published will not start sending that until a subscriber is discovered, the expected output both for publishers and subscriber is a first displayed message acknowledging the match, followed by the amount of samples sent or received until Ctrl+C is pressed.
```

Comment:
```suggestion
Regardless of which application is run first, since the publisher will not start sending data until a subscriber is discovered, the expected output both for publishers and subscribers is a first displayed message acknowledging the match, followed by the amount of samples sent or received until Ctrl+C is pressed.
```